lib/fqn/fqn.go

```diff
@@ -0,0 +1,250 @@
+package util
+
+import (
+	"errors"
+	"fmt"
+	"regexp"
+	"strings"
+)
+
+type FullyQualifiedResourceMappingGroup struct {
+	Fqn       string
+	Namespace string
+	GroupName string
+}
+
+type FullyQualifiedRegisteredResourceValue struct {
+	Fqn   string
+	Name  string
+	Value string
+}
+
+// Structs and regexes for attribute FQNs
+type FullyQualifiedAttribute struct {
+	Fqn       string
+	Namespace string
+	Name      string
+	Value     string
+}
+
+var (
+	ErrInvalidFQNFormat = errors.New("error: invalid FQN format")
+
+	// Regex for attribute value FQN format: https://<namespace>/attr/<name>/value/<value>
+	// The $ at the end ensures no extra segments after value
+	attributeValueFQNRegex = regexp.MustCompile(
+		`^https:\/\/(?<namespace>[^\/]+)\/attr\/(?<name>[^\/]+)\/value\/(?<value>[^\/]+)$`,
+	)
+
+	// Regex for attribute definition FQN format: https://<namespace>/attr/<name>
+	// The $ at the end ensures no extra segments after name
+	attributeDefinitionFQNRegex = regexp.MustCompile(
+		`^https:\/\/(?<namespace>[^\/]+)\/attr\/(?<name>[^\/]+)$`,
+	)
+
+	// Regex for just namespace: https://<namespace>
+	// Only match exactly https:// followed by a namespace with no forward slashes
+	namespaceOnlyRegex = regexp.MustCompile(
+		`^https:\/\/(?<namespace>[^\/]+)$`,
+	)
+
+	// protovalidate already validates the FQN format in the service request
+	// for parsing purposes, we can just look for any non-whitespace characters
+	// e.g. should be in format of "https://<namespace>/resm/<group name>"
+	resourceMappingGroupFqnRegex = regexp.MustCompile(
+		`^https:\/\/(?<namespace>\S+)\/resm\/(?<groupName>\S+)$`,
```

Comment:
If the other regexes are going to use `[^\/]+` to match on a value with any character but `/`, then might as well make it consistent in this and the registered resources regexes.

Reply:
pushed up a commit to address https://github.com/opentdf/platform/pull/2148/commits/ec49870770cd0095060de4cfecf15f21c081b33d